library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -61,25 +70,48 @@ public void setup(PortContext context)
     }
   };
 
-  @OutputPortFieldAnnotation(optional = true)
   public transient DefaultOutputPort<INPUT> err = new DefaultOutputPort<INPUT>();
```

Comment:
Would it be good to have default as  emitting same input to error port and logging error.
If transformation is needed, like you suggested, users can add/override additional port for processedError or connect another operator for error processing to the default error port, whichever applies to specific scenario.